### PR TITLE
Add Submariner UI for Azure cloud

### PR DESCRIPTION
**Pull request by aswinsuryan**
* Added text boxes to accept azure credentials
* The submariner config will created with the values provide

Fixes: [backlog#25786](https://github.com/stolostron/backlog/issues/25786)

Signed-off-by: Aswin Suryanarayanan <aswinsuryan@gmail.com>

### CODE REVIEW

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -2090,7 +2090,8 @@
     "submariner.install.form.gateways.labelHelp": "The number of worker nodes  that will be used to deploy the Submariner gateway component on the managed cluster. If the value is greater than 1, the Submariner gateway HA will be enabled automatically. (default 1)",
     "submariner.install.form.gateways.placeholder": "Enter gateway count",
     "submariner.install.form.instancetype": "Instance type",
-    "submariner.install.form.instancetype.labelHelp": "The Amazon Web Services EC2 instance type of the gateway node that will be created on the managed cluster. (default c5d.large)",
+    "submariner.install.form.instancetype.labelHelp.aws": "The Amazon Web Services EC2 instance type of the gateway node that will be created on the managed cluster. (default c5d.large)",
+    "submariner.install.form.instancetype.labelHelp.az": "The Azure instance type of the gateway node that will be created on the managed cluster. (default Standard_F8s_v2)",
```
Comment (sridhargaddam) [inline frontend/public/locales/en/translation.json:2094]: Should this be "Standard_F4s_v2"?

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 2]
```diff
@@ -643,12 +666,109 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
                                         cluster.provider === Provider.gcp && providerSecretMap[clusterName] === null,
                                     isSecret: true,
                                 },
+                                {
+                                    id: 'baseDomainResourceGroupName',
+                                    type: 'Text',
+                                    label: t('Base domain resource group name'),
+                                    placeholder: t('Enter your base domain resource group name'),
+                                    labelHelp: t(
+                                        'Azure Resources Groups are logical collections of virtual machines, storage accounts, virtual networks, web apps, databases, and/or database servers. Typically, users group related resources for an application, divided into groups for production and non-production.'
```
Comment (sridhargaddam) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:675]:
```suggestion
                                        'Azure Resource Groups are logical collections of virtual machines, storage accounts, virtual networks, web apps, databases, and/or database servers. Typically, users group related resources for an application, divided into groups for production and non-production.'
```

File: frontend/public/locales/en/translation.json (hunk 1 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: ya that is right, this value was something I used for testing , Changed to  Standard_F4s_v2

**Comment by skitt** [comment]
CI is failing with

```
 [frontend] [tsc] src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx(353,58): error TS2339: Property 'azure' does not exist on type '{ aws?: { instanceType: string; } | undefined; gateways?: number | undefined; }'.
```

**Comment by nyechiel** [comment]
@KevinFCormier can someone from the Console team provide review?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
@aswinsuryan Looks pretty good. Thanks for including both Submariner developers and someone from `console` squad (me). I have reviewed in terms of best practices for the `console` repo, and I will leave verification of the Submariner business logic to others.

Please note you will need to address the typescript error that is showing up in the `lint-check-unittests-sonarcloud` job.

Once that is fixed, you will probably run into a Quality Gate problem. Code coverage needs to be 70% on all new code, so you may need to add some unit testing.

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,544 @@
 {
     "name": "@stolostron/console",
     "version": "0.0.1",
-    "lockfileVersion": 1,
+    "lockfileVersion": 2,
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply package-lock.json]: Done.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 4]
```diff
@@ -643,12 +681,108 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
                                         cluster.provider === Provider.gcp && providerSecretMap[clusterName] === null,
                                     isSecret: true,
                                 },
+                                {
+                                    id: 'baseDomainResourceGroupName',
+                                    type: 'Text',
+                                    label: t('Base domain resource group name'),
+                                    placeholder: t('Enter your base domain resource group name'),
+                                    labelHelp: t(
+                                        'Azure Resource Groups are logical collections of virtual machines, storage accounts, virtual networks, web apps, databases, and/or database servers. Typically, users group related resources for an application, divided into groups for production and non-production.'
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:690]: I see it generated. now

**Comment by aswinsuryan** [comment]
> @aswinsuryan Looks pretty good. Thanks for including both Submariner developers and someone from `console` squad (me). I have reviewed in terms of best practices for the `console` repo, and I will leave verification of the Submariner business logic to others.
> 
> Please note you will need to address the typescript error that is showing up in the `lint-check-unittests-sonarcloud` job.
> 
> Once that is fixed, you will probably run into a Quality Gate problem. Code coverage needs to be 70% on all new code, so you may need to add some unit testing.

Thanks for the review,  fixed the typescript error, but now it is failing due to another reason  I think is the unit test issue you mentioned. I will work on it.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (nyechiel) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:690]: @oafischer (docs) might be able to help.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx [hunk 5]
```diff
@@ -123,7 +123,7 @@ export function EditSubmarinerConfigModal(props: EditSubmarinerConfigModalProps)
                                 id="aws-instance-type"
                                 label={t('submariner.install.form.instancetype')}
                                 placeholder={t('submariner.install.form.instancetype.placeholder')}
-                                labelHelp={t('submariner.install.form.instancetype.labelHelp')}
+                                labelHelp={t('submariner.install.form.instancetype.labelHelp.az')}
```
Comment (vthapar) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx:126]: Shouldn't this be `.aws` ?

File: frontend/public/locales/en/translation.json [hunk 6]
```diff
@@ -153,6 +153,7 @@
     "Automations": "Automations",
     "Autoscaling": "Autoscaling",
     "Available cluster types": "Available cluster types",
+    "Azure Resource Groups are logical collections of virtual machines, storage accounts, virtual networks, web apps, databases, and/or database servers. Typically, users group related resources for an application, divided into groups for production and non-production.": "Azure Resource Groups are logical collections of virtual machines, storage accounts, virtual networks, web apps, databases, and/or database servers. Typically, users group related resources for an application, divided into groups for production and non-production.",
```
Comment (vthapar) [inline frontend/public/locales/en/translation.json:156]: Why is this repeated?

**Review by vthapar — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (oafischer) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:690]: I can take a look, just send me a link once the strings are added.

File: frontend/public/locales/en/translation.json (hunk 6 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:156]: this is an autogenerated code, it is a key-value pair.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx (hunk 5 above)
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx]: oh ya , good catch , fixed it.

**Comment by KevinFCormier** [comment]
@aswinsuryan Looks great! I am ready to approve once you have an approved review from a teammate and the strings have been reviewed by the doc team. CC @oafischer

**Comment by nyechiel** [comment]
> @aswinsuryan Looks great! I am ready to approve once you have an approved review from a teammate and the strings have been reviewed by the doc team. CC @oafischer

@sridhargaddam @vthapar can you please review/approve when you have some time? thanks!

File: frontend/src/resources/submariner-config.ts [hunk 7]
```diff
@@ -53,4 +63,5 @@ export const submarinerConfigDefault: SubmarinerConfigDefaults = {
     cableDriver: CableDriver.libreswan,
     gateways: 1,
     awsInstanceType: 'c5d.large',
+    azInstanceType: 'Standard_F4s_v2',
```
Comment (sridhargaddam) [inline frontend/src/resources/submariner-config.ts:66]: nit: can we use azure? (i.e., azureInstanceType)

**Review by vthapar — DISMISSED** [review]

File: frontend/public/locales/en/translation.json [hunk 8]
```diff
@@ -158,6 +158,7 @@
...
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:161]: _Typically, users group related resources for an application_

What is this trying to say?

File: frontend/public/locales/en/translation.json [hunk 9]
```diff
@@ -2115,7 +2116,8 @@
     "submariner.install.form.gateways.labelHelp": "The number of worker nodes  that will be used to deploy the Submariner gateway component on the managed cluster. If the value is greater than 1, the Submariner gateway HA will be enabled automatically. (default 1)",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2116]: Change to:

The number of worker nodes used to deploy the Submariner gateway component on the managed cluster. If the value is greater than 1, the Submariner gateway HA is enabled automatically (default 1).

Also, what is HA?

File: frontend/public/locales/en/translation.json [hunk 10]
```diff
@@ -2115,7 +2116,8 @@
...
     "submariner.install.form.gateways.placeholder": "Enter gateway count",
     "submariner.install.form.instancetype": "Instance type",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2118]: _Enter gateway count_ has a prompt but _Instance type_ doesn't. Do they both need a prompt?

File: frontend/public/locales/en/translation.json [hunk 11]
```diff
@@ -2115,7 +2116,8 @@
...
-    "submariner.install.form.instancetype.labelHelp": "The Amazon Web Services EC2 instance type of the gateway node that will be created on the managed cluster. (default c5d.large)",
+    "submariner.install.form.instancetype.labelHelp.aws": "The Amazon Web Services EC2 instance type of the gateway node that will be created on the managed cluster. (default c5d.large)",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2119]: Not sure how it is in other UI instances but typically we would have the parenthesis before the period. I changed it in an example above but just do whatever keeps it consistent across the UI.

File: frontend/public/locales/en/translation.json [hunk 12]
```diff
@@ -2115,7 +2116,8 @@
...
+    "submariner.install.form.instancetype.labelHelp.azure": "The Azure instance type of the gateway node that will be created on the managed cluster. (default Standard_F4s_v2)",
     "submariner.install.form.instancetype.placeholder": "Select instance type",
     "submariner.install.form.nattenable": "Enable NAT-T",
     "submariner.install.form.nattenable.labelHelp": "Submariner can use NAT Traversal for IPsec tunnels between clusters. Set this to true to enable NAT Traversal and false otherwise (default true)",
```
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2123]: Change to:

Set the value to true to enable NAT Traversal and false to disable NAT Traversal (default true).

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 13]
```diff
@@ -643,12 +682,108 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                                    isRequired:
+                                        cluster.provider === Provider.azure && providerSecretMap[clusterName] === null,
+                                },
+                                {
+                                    id: 'clientId',
+                                    type: 'Text',
+                                    label: t('Client ID'),
+                                    placeholder: t('Enter your client ID'),
+                                    labelHelp: t(
+                                        "Your client ID. This value is generated as the 'appId' property when you create a service principal with the command: 'az ad sp create-for-rbac --role Contributor --name <service_principal> --scopes <list_of_scopes>'."
```
Comment (oafischer) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:710]: -> ...with the following command:

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 14]
```diff
@@ -643,12 +682,108 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                                        cluster.provider === Provider.azure && providerSecretMap[clusterName] === null,
+                                    isSecret: true,
+                                },
+                                {
+                                    id: 'subscriptionId',
+                                    type: 'Text',
+                                    label: t('Subscription ID'),
+                                    placeholder: t('Enter your subscription ID'),
+                                    labelHelp: t(
+                                        "Your subscription ID. This is the value of the 'id' property in the output of the command: 'az account show'"
```
Comment (oafischer) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:749]: -> ...with the following command:

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 15]
```diff
@@ -643,12 +682,108 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                                    isRequired:
+                                        cluster.provider === Provider.azure && providerSecretMap[clusterName] === null,
+                                },
+                                {
+                                    id: 'tenantId',
+                                    type: 'Text',
+                                    label: t('Tenant ID'),
+                                    placeholder: t('Enter your tenant ID'),
+                                    labelHelp: t(
+                                        "Your tenant ID. This is the value of the 'tenantId' property in the output of the command: 'az account show'"
```
Comment (oafischer) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:768]: -> ...of the following command:

**Review by oafischer — CHANGES_REQUESTED** [review]
Left a few comments on the strings.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 14 above)
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx]: done.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx (hunk 15 above)
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx]: Done.

File: frontend/public/locales/en/translation.json (hunk 8 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:161]: This is not related to Submariner, I am not sure about the context either.

File: frontend/public/locales/en/translation.json (hunk 9 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: Done, HA is high availablity

File: frontend/public/locales/en/translation.json (hunk 10 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:2118]: Sorry , I am not clear about this, I don't see a prompt here

File: frontend/public/locales/en/translation.json (hunk 11 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: I made the period at the end for all Submariner strings

File: frontend/public/locales/en/translation.json (hunk 12 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: Done

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx [hunk 16]
```diff
@@ -643,12 +682,108 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
...
+                                    isRequired:
+                                        cluster.provider === Provider.azure && providerSecretMap[clusterName] === null,
+                                },
+                                {
+                                    id: 'clientSecret',
+                                    type: 'Text',
+                                    label: t('Client secret'),
+                                    placeholder: t('Enter your client secret'),
+                                    labelHelp: t(
+                                        "Your client password. This value is generated as the 'password' property when you create a service principal with the command: 'az ad sp create-for-rbac --role Contributor --name <service_principal> --scopes <list_of_scopes>'."
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx]: Done,

**Comment by aswinsuryan** [comment]
@oafischer Thanks for your review. I addressed it. Have a query about on of them.

File: frontend/public/locales/en/translation.json (hunk 10 above)
Comment (oafischer) [reply frontend/public/locales/en/translation.json:2118]: The prompt here is "Enter"

File: frontend/public/locales/en/translation.json (hunk 8 above)
Comment (oafischer) [reply frontend/public/locales/en/translation.json:161]: OK, either way the sentences don't make sense so give me a minute and I can provide a suggestion to improve them.

File: frontend/public/locales/en/translation.json (hunk 10 above)
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: oh ok, I added in the prompt in instance type as well,

File: frontend/public/locales/en/translation.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:161]: When I checked the usages of the string it is used only for baseDomainGroup in submarinerInstall page and the string you mentioned above makes sense. I changed it.

**Comment by aswinsuryan** [comment]
> Left one more suggestion for the Azure Resource Groups string. Not sure who would be the right person to review. Thanks for making the other changes @aswinsuryan!

@oafischer I checked usages for the string and is used for Azure baseDomainGroup and the new string you provided is appropriate. I changed it in all the places where I found it is used

**Comment by oafischer** [comment]
Thanks @aswinsuryan! Approving from a doc/writing perspective.

**Review by oafischer — APPROVED** [review]

**Pull request merged**